test/unit/test_1D_linear_interpolator.cpp

```diff
@@ -0,0 +1,21 @@
+#include "linear_interpolator_1D.hpp"
+#include <CL/sycl.hpp>
+#include <gtest/gtest.h>
+#include <mpi.h>
+#include <neso_particles.hpp>
+#include <vector>
+
+using namespace NESO;
+using namespace NESO::Particles;
+
+TEST(InterpolatorTest, 1DLinear) {
+  auto sycl_target = std::make_shared<SYCLTarget>(0, MPI_COMM_WORLD);
+  std::vector<double> test_input = {1.5, 2.5};
+  std::vector<double> &test_input_ref = test_input;
+  std::vector<double> test_output(test_input.size());
+  std::vector<double> &test_output_ref = test_output;
+  LinearInterpolator1D({0, 1, 2, 3}, {0, 1, 2, 3}, sycl_target)
```

Comment:
These input data test points are in a straight line - so any indexing bugs relating to the gradient are invisible to the test as all the mesh bins have the same gradient.

Reply:
i.e. piecewise linear segments are fine for a known exact result but the gradients should change between the bins.